addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -116,5 +116,58 @@ const PopupWidget = publicWidget.Widget.extend({
 
 publicWidget.registry.popup = PopupWidget;
 
+// Extending the popup widget with cookiebar functionality.
+// This allows for refusing optional cookies for now and can be
+// extended to picking which cookies categories are accepted.
+publicWidget.registry.cookies_bar = PopupWidget.extend({
+    selector: '#website_cookies_bar',
+    cookieDurationDays: 365,
+    events: Object.assign({}, PopupWidget.prototype.events, {
+        'click #cookies-consent-essential': '_onConsentEssentialClick',
+        'click #cookies-consent-all': '_onConsentAllClick',
+    }),
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @param acceptOptional Whether optional cookies were accepted or not
+     */
+    _onAcceptClick(acceptOptional) {
+        const acceptedCookieTypes = JSON.parse(get_cookie('accepted_cookie_types') || '{}');
+        Object.assign(acceptedCookieTypes, {'required': true, 'optional': acceptOptional});
+        set_cookie('accepted_cookie_types',
+            JSON.stringify(acceptedCookieTypes),
+            this.cookieDurationDays * 24 * 60 * 60, 'required');
+        // Prevent the modal from reopening again.
+        this._onHideModal();
+    },
+    /**
+     * @private
+     */
+    _onConsentAllClick() {
+        this._onAcceptClick(true);
+    },
+    /**
+     * @private
+     */
+    _onConsentEssentialClick() {
+        this._onAcceptClick(false);
+    },
+    /**
+     * @private
+     */
+    _onHideModal() {
+        this._super.apply(this, arguments);
+        if (!get_cookie('accepted_cookie_types')) {
```

Comment:
How can someone close that modal without going through the button? It should be the only way to close it

Reply:
You can close the modal by pressing "escape" - this currently is interpreted by the cookies bar as "I'll respond later" (and you get the popup again on the next page load).

Reply:
It is not supposed to be the case (as opposed to a regular popup), how did you do that? I just tried in 15 and 16 and your branch and I can't close it with escape regardless of the layout and backdrop option.

Reply:
Indeed, it seems to not be the case anymore. Strange, that behavior was even agreed on with the PO.
I remove that function (will be in the next push).